core-api/src/main/java/com/optimizely/ab/OptimizelyUserContext.java

```diff
@@ -0,0 +1,186 @@
+        return userId;
+    }
+
+    public Map<String, Object> getAttributes() {
+        return new HashMap<String, Object>(attributes);
+    }
+
+    public Optimizely getOptimizely() {
+        return optimizely;
+    }
+
+    /**
+     * Set an attribute for a given key.
+     *
+     * @param key An attribute key
+     * @param value An attribute value
+     */
+    public void setAttribute(@Nonnull String key, @Nonnull Object value) {
+        attributes.put(key, value);
+    }
+
+    /**
+     * Returns a decision result ({@link OptimizelyDecision}) for a given flag key and a user context, which contains all data required to deliver the flag.
+     * <ul>
+     * <li>If the SDK finds an error, it’ll return a decision with <b>null</b> for <b>variationKey</b>. The decision will include an error message in <b>reasons</b>.
+     * </ul>
+     * @param key A flag key for which a decision will be made.
+     * @param options A list of options for decision-making.
+     * @return A decision result.
+     */
+    public OptimizelyDecision decide(@Nonnull String key,
+                                     @Nonnull List<OptimizelyDecideOption> options) {
+        return optimizely.decide(this, key, options);
+    }
+
+    /**
+     * Returns a decision result ({@link OptimizelyDecision}) for a given flag key and a user context, which contains all data required to deliver the flag.
+     *
+     * @param key A flag key for which a decision will be made.
+     * @return A decision result.
+     */
+    public OptimizelyDecision decide(String key) {
+        return optimizely.decide(this, key, Collections.emptyList());
+    }
+
+    /**
+     * Returns a key-map of decision results ({@link OptimizelyDecision}) for multiple flag keys and a user context.
+     * <ul>
+     * <li>If the SDK finds an error for a key, the response will include a decision for the key showing <b>reasons</b> for the error.
+     * <li>The SDK will always return key-mapped decisions. When it can not process requests, it’ll return an empty map after logging the errors.
+     * </ul>
+     * @param keys A list of flag keys for which decisions will be made.
+     * @param options A list of options for decision-making.
+     * @return All decision results mapped by flag keys.
+     */
+    public Map<String, OptimizelyDecision> decideForKeys(@Nonnull List<String> keys,
+                                                         @Nonnull List<OptimizelyDecideOption> options) {
+        return optimizely.decideForKeys(this, keys, options);
+    }
+
+    /**
+     * Returns a key-map of decision results for multiple flag keys and a user context.
+     *
+     * @param keys A list of flag keys for which decisions will be made.
+     * @return All decision results mapped by flag keys.
+     */
+    public Map<String, OptimizelyDecision> decideForKeys(@Nonnull List<String> keys) {
+        return optimizely.decideForKeys(this, keys, Collections.emptyList());
```

Comment:
call decideForKeys of this class.

Reply:
Will keep it to save one-step indirection.